homeassistant/components/overkiz/coordinator.py

```diff
@@ -150,14 +104,87 @@ async def _get_devices(self) -> dict[str, Device]:
         _LOGGER.debug("Fetching all devices and state via /setup/devices")
         return {d.device_url: d for d in await self.client.get_devices(refresh=True)}
 
-    def places_to_area(self, place: Place) -> dict[str, str]:
-        """Convert places with sub_places to a flat dictionary."""
+    def _places_to_area(self, place: Place) -> dict[str, str]:
+        """Convert places with sub_places to a flat dictionary [placeoid, label])."""
         areas = {}
         if isinstance(place, Place):
             areas[place.oid] = place.label
 
         if isinstance(place.sub_places, list):
             for sub_place in place.sub_places:
-                areas.update(self.places_to_area(sub_place))
+                areas.update(self._places_to_area(sub_place))
 
         return areas
+
+
+@EVENT_HANDLERS.register(EventName.DEVICE_AVAILABLE)
+async def on_device_available(
+    coordinator: OverkizDataUpdateCoordinator, event: Event
+) -> None:
+    """Handle device available event."""
+    if event.device_url:
+        coordinator.devices[event.device_url].available = True
+
+
+@EVENT_HANDLERS.register((EventName.DEVICE_UNAVAILABLE, EventName.DEVICE_DISABLED))
```

Comment:
The instance we pass to `register` will be the key in the registry dict. It won't work to use a tuple as key unless we use a tuple when we look up the handler.

Reply:
e1f13a79d65d6c97e776741f871c07d3d200184c